cmd/kubeadm/app/util/etcd/etcd_test.go

```diff
@@ -84,6 +84,64 @@ spec:
       type: DirectoryOrCreate
     name: etcd-certs
 status: {}
+`
+	secureExposedEtcdPod = `# generated by kubeadm v1.10.0
+apiVersion: v1
+kind: Pod
+metadata:
+  annotations:
+    scheduler.alpha.kubernetes.io/critical-pod: ""
+  creationTimestamp: null
+  labels:
+    component: etcd
+    tier: control-plane
+  name: etcd
+  namespace: kube-system
+spec:
+  containers:
+  - command:
+    - etcd
+    - --advertise-client-urls=https://10.0.5.5:2379
+    - --data-dir=/var/lib/etcd
+    - --peer-key-file=/etc/kubernetes/pki/etcd/peer.key
+    - --peer-trusted-ca-file=/etc/kubernetes/pki/etcd/ca.crt
+    - --listen-client-urls=https://[::0:0]:2379
```

Comment:
`https://[::]:2379`

Reply:
This is also intentional.
It forces us to use `net.ParseIP()` because these are all equivalent:
```
0:0:0:0:0:0:0:0
::0:0:0:0:0
::0:0
::0
::
```

Reply:
I understand that those are equivalent, but what is the reason behind that particular notation ?

Reply:
It prevents code that just does a string match on `::` or `::0` instead of parsing the IP and doing real equivalence.

If you'd like I can expand the test cases, but I figured that if I just add one, I should pick an edge case.